### DISCUSSION TITLE
Window rule to auto-consume child windows into the same column

### DISCUSSION THREAD

**Discussion by Nairou** [YaLTeR/niri#410] [Ideas]
I've noticed several applications that open separate windows, such as a save confirmation dialog or a tool window of some sort. Currently, all of these windows appear next to the main window in their own column, sometimes pushing the main window partly off-screen.

It might be useful to have a window rule that could be applied to certain applications, such that all windows get added to the current column rather than starting a new column. Essentially performing an automatic `consume-window-into-column` when the child window is opened.

**Comment by YaLTeR** [comment]
Hmm. For dialogs specifically, ideally they should open floating, either on the floating layer when that happens, or somehow attached to the parent window if modal. That reduces the number of cases where a window rule for consume would be useful.

**Comment by Nairou** [comment]
The cases I keep running into are Sublime Text 4, Blender, and Meld. All windows for File Open, or Save confirmation, end up showing as separate windows in a new column.

If there was a way to force these to be treated as a dialog, that would certainly be ideal.

**Comment by y0usaf** [comment]
This is now implemented in PR #3508.
